src/decode/lzma2.rs

```diff
@@ -103,15 +112,15 @@ where
     );
 
     if reset_dict {
-        decoder.output.reset()?;
+        accum.reset()?;
     }
 
     if reset_state {
-        let lc: u32;
-        let lp: u32;
-        let mut pb: u32;
+        let new_props = if reset_props {
+            let lc: u32;
+            let lp: u32;
+            let mut pb: u32;
```

Comment:
Please check and fix the lints in this function as reported by Clippy: https://github.com/gendx/lzma-rs/runs/6790343456

Reply:
fixed: https://github.com/gendx/lzma-rs/commit/163b034954e2c0295a8e25c5567559fdccce2470#diff-aec280556b32db2cd7e0d9b67090c0833582fa5e5dac272202f84adcfd64e887R124